src/main/java/org/jasig/portlet/announcements/mvc/portlet/display/AnnouncementsViewController.java

```diff
@@ -190,6 +178,46 @@ public String mainView(
     return viewNameSelector.select(request, "displayAnnouncements");
   }
 
+  private List[] getLists(PortletRequest request) throws PortletException {
+    final String userId = userIdService.getUserId(request);
+
+    // create a new announcement list
+    List<Announcement> announcements1 = new ArrayList<>();
+    List<Announcement> emergencyAnnouncements1 = new ArrayList<>();
+
+    // fetch the user's topic subscription from the database
+    List<TopicSubscription> myTopics = tss.getTopicSubscription(request);
+
+    // add all the published announcements of each subscribed topic to the announcement list
+    // to emergency announcements into their own list
+    for (TopicSubscription ts : myTopics) {
+      if (ts.getSubscribed() && ts.getTopic().getSubscriptionMethod() != Topic.EMERGENCY) {
+        announcements1.addAll(ts.getTopic().getPublishedAnnouncements());
```

Comment:
https://www.mkyong.com/java8/java-8-streams-filter-examples/

Reply:
Well, I thought that the stream() version would be longer and harder to understand; however, after digging in and refining, I was able to remove other lines to make the stream() version smaller. Thanks for the push. This was also the first time I used flatMap() in Java.